x-pack/legacy/plugins/lens/public/editor_frame_plugin/editor_frame/editor_frame.tsx

```diff
@@ -149,28 +180,37 @@ export function EditorFrame(props: EditorFrameProps) {
         dataPanel={
           <DataPanelWrapper
             datasourceMap={props.datasourceMap}
```

Comment:
This DataPanelWrapper appears to be duplicated further down (on line 240). Not sure if it's a problem. Just pointing it out.

Reply:
Not really, it's just not rendering the config panel and the workspace as long as the datasource is loading. We are probably going to change this when refining the UI (maybe a loading spinner for the whole app or something like that?). Till then I simplified the UI a bit by moving the condition for the other panels into a single JSX tree.